sdk/tests/conformance/glsl/bugs/bool-type-cast-bug.html

```diff
@@ -0,0 +1,317 @@
+        return;
+    }
+    wtu.setupUnitQuad(gl);
+
+    debug("");
+    var program = wtu.setupProgram(gl, ["vshader-int1", "fshader-int1"], ["aPosition"]);
+    if (!program) {
+        testFailed("Fail to set up program");
+    } else {
+        var uniformLoc = gl.getUniformLocation(program, 'bvalue');
+        debug("Testing vertex shader int(false)");
+        gl.uniform1i(uniformLoc, 0);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [0, 255, 0, 255]);
+        debug("Testing vertex shader int(true)");
+        gl.uniform1i(uniformLoc, 1);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [255, 0, 0, 255]);
+        gl.deleteProgram(program);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Should be no errors from testing");
+    }
+
+    debug("");
+    program = wtu.setupProgram(gl, ["vshader-int2", "fshader-int2"], ["aPosition"]);
+    if (!program) {
+        testFailed("Fail to set up program");
+    } else {
+        var uniformLoc = gl.getUniformLocation(program, 'bvalue');
+        debug("Testing fragment shader int(false)");
+        gl.uniform1i(uniformLoc, 0);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [0, 255, 0, 255]);
+        debug("Testing fragment shader int(true)");
+        gl.uniform1i(uniformLoc, 1);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [255, 0, 0, 255]);
+        gl.deleteProgram(program);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Should be no errors from testing");
+    }
+
+    debug("");
+    program = wtu.setupProgram(gl, ["vshader-float1", "fshader-float1"], ["aPosition"]);
+    if (!program) {
+        testFailed("Fail to set up program");
+    } else {
+        var uniformLoc = gl.getUniformLocation(program, 'bvalue');
+        debug("Testing vertex shader float(false)");
+        gl.uniform1i(uniformLoc, 0);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [0, 255, 0, 255]);
+        debug("Testing vertex shader float(true)");
+        gl.uniform1i(uniformLoc, 1);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [255, 0, 0, 255]);
+        gl.deleteProgram(program);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Should be no errors from testing");
+    }
+
+    debug("");
+    program = wtu.setupProgram(gl, ["vshader-float2", "fshader-float2"], ["aPosition"]);
+    if (!program) {
+        testFailed("Fail to set up program");
+    } else {
+        var uniformLoc = gl.getUniformLocation(program, 'bvalue');
+        debug("Testing fragment shader float(false)");
+        gl.uniform1i(uniformLoc, 0);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [0, 255, 0, 255]);
+        debug("Testing fragment shader float(true)");
+        gl.uniform1i(uniformLoc, 1);
+        wtu.drawUnitQuad(gl);
+        wtu.checkCanvas(gl, [255, 0, 0, 255]);
+        gl.deleteProgram(program);
+        wtu.glErrorShouldBe(gl, gl.NO_ERROR, "Should be no errors from testing");
+    }
+
+    if (wtu.getDefault3DContextVersion() > 1) {
```

Comment:
It seems a little weird to me to put a WebGL2-specific test in conformance/, but I don't see any problem with it. @kenrussell, can you comment?